extensions/GradioNotebook/gradionotebook/frontend/GradioWorkbook.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  AIConfigEditor,
+  type AIConfigCallbacks,
+  type LogEvent,
+  type LogEventData,
+} from "@lastmileai/aiconfig-editor";
+import type { AIConfig } from "aiconfig";
+import APITokenInput from "./APITokenInput";
+import { Flex, MantineProvider, type MantineTheme } from "@mantine/core";
+import WorkbookInfoAlert from "./WorkbookInfoAlert";
+import { useMemo } from "react";
+import { useCallback, useEffect } from "react";
+import { datadogLogs } from "@datadog/browser-logs";
+import * as uuid from "uuid";
+
+type Props = {
+  aiconfig: AIConfig;
+  editorCallbacks: AIConfigCallbacks;
+  onSetApiToken: (apiToken: string) => Promise<void>;
+  themeMode: "light" | "dark" | "system";
+};
+
+const MODE = "gradio";
+
+export default function GradioWorkbook(props: Props) {
+  // AIConfigEditor handles dynamic system theme switching by default, so only
+  // pass dark or light override
+  const themeMode = props.themeMode === "system" ? undefined : props.themeMode;
+
+  const theme = useMemo(
```

Comment:
Ah, note to self -- we should be able to remove this theme stuff here now that it's updated in GradioTheme to have proper specificity.

To be done in a later PR